From 6eba7ada5a0dc3a43840025df8444b4b113732e3 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 24 May 2019 09:31:44 +0800
Subject: [PATCH] ASoC: rk3308_codec: increase some delays to be stabe enabling
 DAC

From the suggest of IP vendor, we need more delays to
adapt to the playing scene in different temperature
environments.

Change-Id: I95abe2bac43de19241092d6e785f28960760dacd
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 sound/soc/codecs/rk3308_codec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index 4f24b84ca3f0..31ebbcbc249c 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -2080,7 +2080,7 @@ static int rk3308_codec_dac_enable(struct rk3308_codec_priv *rk3308)
 			   RK3308_DAC_BUF_REF_R_EN);
 
 	/* Waiting the stable reference voltage */
-	udelay(50);
+	mdelay(1);
 
 	if (rk3308->dac_output == DAC_HPOUT ||
 	    rk3308->dac_output == DAC_LINEOUT_HPOUT) {
@@ -2115,7 +2115,7 @@ static int rk3308_codec_dac_enable(struct rk3308_codec_priv *rk3308)
 			   RK3308_DAC_R_HPMIX_EN);
 
 	/* Waiting the stable HPMIX */
-	udelay(50);
+	mdelay(1);
 
 	/* Step 06. Reset HPMIX and recover HPMIX gains */
 	regmap_update_bits(rk3308->regmap, RK3308_DAC_ANA_CON13,
-- 
2.35.3

